File: 3090-maximum-length-substring-with-two-occurrences/3090-maximum-length-substring-with-two-occurrences.java
class Solution {
    public int maximumLengthSubstring(String s) {
        
        Map<Character,Integer> count = new HashMap<>();
        int n = s.length();
        int l = 0;
        int best = 0;
        for (int i = 0; i < n; i++) {
            char c = s.charAt(i);
            count.put(c, count.getOrDefault(c, 0) + 1);

            while (l < n && count.get(c) > 2) {
                char c_left = s.charAt(l);
                count.put(c_left, count.get(c_left) - 1);
                l++;
            }

        best = Math.max(best, i - l + 1);
        }
        return best;
    }
}

File: 3090-maximum-length-substring-with-two-occurrences/3090-maximum-length-substring-with-two-occurrences.py
class Solution:
    def maximumLengthSubstring(self, s: str) -> int:
        
        n = len(s)
        l = 0
        char_freq = defaultdict(int)
        best = 0
        for r in range(n):
            ch = s[r]

            char_freq[ch] += 1
            while char_freq[ch] > 2:
                ch_left = s[l]
                char_freq[ch_left] -= 1
                l += 1
            

            
            best = max(best, r - l + 1)
        return best


